packages/messaging/lib/index.js

```diff
@@ -154,12 +155,34 @@ class FirebaseMessagingModule extends FirebaseModule {
     return this.native.getIsHeadless();
   }
 
-  getToken() {
-    return this.native.getToken();
+  getToken({ projectId, senderId } = {}) {
+    if (!isUndefined(projectId) && !isString(projectId)) {
+      throw new Error("firebase.messaging().getToken(*) 'projectId' expected a string.");
+    }
+
+    if (!isUndefined(senderId) && !isString(senderId)) {
+      throw new Error("firebase.messaging().getToken(*) 'senderId' expected a string.");
+    }
+
+    return this.native.getToken(
+      projectId || this.app.options.projectId,
```

Comment:
So once it's not the projectId but the appName that the getInstance expects, you should pass this.app.name, I've tested locally here and it worked

Reply:
Great! Thanks for checking that, and I agree with you as the error message was clearly suggesting app names as the possible values that could be sent in. Might be close on this whole PR with that change to `appName` as a TokenOptions object property plus the typo fix in firebase-ios-sdk method name above

Reply:
@mikehardy @FranciscoPDNeto HUGE Thank You for the suggestions in here. I believe I've applied them all, and see the fixes reflected, BUT there seem to be more test errors that are even more elusive. Let me know if y'all understand the new issues?

Reply:
Yes, this test that failed on iOS is failing here at my local, I saw that @mikehardy put an if to skip this into CI, but it seems that is running anyway (Here it was the only test that failed)

Reply:
I was thinking that maybe this failing can be related to the latest updates on main branch, because yesterday when I tried it was not failing (using the fork that was kind of 30 commits behind the main). On iOS it was failing another test, but not that one, and the android tests were passing

Reply:
I've got a commit locally that fixes the iOS storage reference thing - it's not related to this PR at all, one of the helper functions I wrote to install storage security rules was async and (classic!) I forgot the await, so it's in a race with usage of the rules and sometime it works (android, sometimes iOS), sometimes it fails (ios sometimes). I'll commit that in a moment. It appears the test suite made it past the auth token part of the test already so this PR is likely good to go, reviewing now